src/db/users.rs

```diff
@@ -6,6 +6,34 @@ use diesel::pg::PgConnection;
 use diesel::prelude::*;
 use serde::Deserialize;
 
+
+pub fn username_exists(conn: &PgConnection, username: &str) -> bool {
+    use crate::schema::users;
+    let n: i64 = users::table
+        .filter(users::username.eq(&username))
+        .count()
+        .get_result(conn)
+        .expect("count username");
+    if n > 0 {
+        return true;
+    }
+    false
+}
+pub fn email_exists(conn: &PgConnection, email: &str) -> bool {
+    use crate::schema::users;
+    let n: i64 = users::table
+        .filter(users::email.eq(&email))
+        .count()
```

Comment:
Also consider using [exists](https://docs.diesel.rs/diesel/dsl/fn.exists.html) to make code even cleaner.